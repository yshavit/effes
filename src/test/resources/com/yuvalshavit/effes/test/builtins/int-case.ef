isZeroA (arg: Int) -> True | False: return case arg of
  IntZero: True
  IntValue: False

isZeroB (arg: Int) -> True | False: return case arg of
  IntValue: False
  IntZero: True

main:
  debugPrint isZeroA 0
  debugPrint isZeroA 1
  debugPrint isZeroB 0
  debugPrint isZeroB 1
