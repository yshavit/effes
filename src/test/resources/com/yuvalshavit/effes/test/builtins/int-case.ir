def isZeroA (arg: IntValue | IntZero) -> False | True:
  ReturnStatement
    CaseExpression
      case:
        VarExpression
          arg: arg
          pos: 0
      of IntZero:
        CtorInvoke
          type: True
      of IntValue (exhaustive):
        CtorInvoke
          type: False
def isZeroB (arg: IntValue | IntZero) -> False | True:
  ReturnStatement
    CaseExpression
      case:
        VarExpression
          arg: arg
          pos: 0
      of IntValue (exhaustive):
        CtorInvoke
          type: False
      of IntZero:
        CtorInvoke
          type: True
