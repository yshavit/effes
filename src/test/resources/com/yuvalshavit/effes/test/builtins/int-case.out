True
False
True
False
>> null
